src/sage/combinat/bijectionist.py

```diff
@@ -0,0 +1,3217 @@
+                i += 1
+                assert i == len(self._solution_cache)
+                yield self._prepare_solution(on_blocks, solution)
+                if get_verbose() >= 2:
+                    print("after vetoing")
+                    self.show(variables=False)
+
+    def _add_solution(self, solution):
+        r"""
+        Add the ``solution`` to the cache and an appropriate
+        veto constraint to the MILP.
+
+        INPUT:
+
+        - ``solution``, a dictionary from the indices of the MILP to
+          Boolean.
+
+        EXAMPLES::
+
+            sage: A = B = ["a", "b"]
+            sage: bij = Bijectionist(A, B)
+            sage: from sage.combinat.bijectionist import _BijectionistMILP
+            sage: bmilp = _BijectionistMILP(bij)
+            sage: bmilp._add_solution({(a, b): a == b for a in A for b in B})
+            sage: bmilp.show()                                                  # random
+            Constraints are:
+                block a: 1 <= x_0 + x_1 <= 1
+                block b: 1 <= x_2 + x_3 <= 1
+                statistics: 1 <= x_1 + x_3 <= 1
+                statistics: 1 <= x_0 + x_2 <= 1
+                veto: x_1 + x_2 <= 1
+            Variables are:
+                x_0: s(a) = b
+                x_1: s(a) = a
+                x_2: s(b) = b
+                x_3: s(b) = a
+
+        """
+        active_vars = [self._x[p, z]
+                       for p in _disjoint_set_roots(self._bijectionist._P)
+                       for z in self._bijectionist._possible_block_values[p]
+                       if solution[(p, z)]]
+        self.milp.add_constraint(sum(active_vars) <= len(active_vars) - 1,
+                                 name="veto")
+        self._solution_cache.append(solution)
+
+    def _is_solution(self, constraint, values):
+        r"""
+        Evaluate the given function at the given values.
+
+        INPUT:
+
+        - ``constraint``, a
+          :class:`sage.numerical.linear_functions.LinearConstraint`.
+
+        - ``values``, a candidate for a solution of the MILP as a
+          dictionary from pairs `(a, z)\in A\times Z` to `0` or `1`,
+          specifying whether `a` is mapped to `z`.
+
+        EXAMPLES::
+
+            sage: A = B = ["a", "b"]
+            sage: bij = Bijectionist(A, B)
+            sage: from sage.combinat.bijectionist import _BijectionistMILP
+            sage: bmilp = _BijectionistMILP(bij)
+            sage: f = bmilp._x["a", "a"] + bmilp._x["b", "a"] >= bmilp._x["b", "b"] + 1
+            sage: v = {('a', 'a'): 1, ('a', 'b'): 0, ('b', 'a'): 1, ('b', 'b'): 1}
+            sage: bmilp._is_solution(f, v)
+            True
+            sage: v = {('a', 'a'): 0, ('a', 'b'): 0, ('b', 'a'): 1, ('b', 'b'): 1}
+            sage: bmilp._is_solution(f, v)
+            False
+        """
+        index_block_value_dict = {}
+        for (p, z), v in self._x.items():
+            variable_index = next(iter(v.dict().keys()))
```

Comment:
```suggestion
            variable_index = next(iter(v.dict()))
```
I think calling `keys()` here is unnecessary, but not 100% sure.